src/gui/editorwidgets/core/qgseditorwidgetregistry.h

```diff
@@ -33,6 +33,35 @@ class QgsVectorLayer;
 
 
 /** \ingroup gui
+ * Holder for the widget type and its configuration.
+ *
+ * @note added in QGIS 3.0
+ */
+class GUI_EXPORT QgsWidgetSetup
```

Comment:
Can this be renamed to start with `QgsEditorWidget` to be inline with the other classes?